src/Application/Commands/ReturnErroneousExcessFees.php

```diff
@@ -0,0 +1,149 @@
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Temporary command to correct fees overcharged in error, so that charities receive an extra
+ * payout covering the difference ASAP.
+ */
+#[AsCommand(
+    name: 'matchbot:fix-fees',
+    description: "Prints or fixes fees accidentally overcharged to some charities",
+)]
+class ReturnErroneousExcessFees extends LockingCommand
+{
+    public function __construct(
+        private DonationRepository $donationRepository,
+        private readonly StripeClient $stripeClient,
+    ) {
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this->addArgument(
+            'mode',
+            InputArgument::REQUIRED,
+            '"check" to print status information only or "fix" to issue fee refunds.'
+        );
+    }
+
+    protected function doExecute(InputInterface $input, OutputInterface $output): int
+    {
+        $mode = $input->getArgument('mode');
+        if (!in_array($mode, ['check', 'fix'], true)) {
+            $output->writeln('Please set the mode to "check" or "fix"');
+            return 1;
+        }
+
+        $donations = $this->donationRepository->findWithFeePossiblyOverchaged();
+
+        $countOfDonationsChecked = count($donations);
+        $countOfDonationsWithDiscrepancy = 0;
+        $countOfDonationsToChange = 0;
+
+        /** @var numeric-string $sumOfFeeDifference */
+        $sumOfFeeDifference = '0.00';
+
+        foreach ($donations as $donation) {
+            $chargeId = $donation->getChargeId();
+            Assertion::notNull($chargeId, "Donation {$donation->getUuid()} has no charge ID");
+            $charge = $this->stripeClient->charges->retrieve($chargeId);
+
+            if (!$this->feeDiffers($donation, $charge) || $charge->status !== 'succeeded') {
+                continue;
+            }
+
+            $countOfDonationsWithDiscrepancy++;
+            $feeId = (string) $charge->application_fee;
+            $fee = $this->stripeClient->applicationFees->retrieve($feeId);
+
+            if ($fee->amount_refunded > 0) {
+                // Might be because the donation itself was fully refunded and the fee reversed. Carry on.
+                $output->writeln("Donation {$donation->getId()} has already had fee refunded");
+                continue;
+            }
+
+            $feeDifference = $fee->amount - $donation->getAmountToDeductFractional();
+
+            if ($donation->getCharityFee() !== $this->getCorrectFees($donation, $charge)->coreFee) {
+                $output->writeln("Donation {$donation->getId()} has a different fee than expected");
+                continue;
+            }
+
+            $countOfDonationsToChange++;
+            $sumOfFeeDifference = bcadd($sumOfFeeDifference, (string) ($feeDifference / 100), 2);
```

Comment:
rename to $sumOfFeeDifferencePounds ?

Reply:
Done in 975a424bb1d2decfcd7f4cd6298666590505b3cf